cmd/influx/task.go

```diff
@@ -3,342 +3,455 @@ package main
 
 		tasks = append(tasks, task)
 	} else {
-		tasks, _, err = s.FindTasks(context.Background(), filter)
+		tasks, _, err = tskSvc.FindTasks(context.Background(), filter)
 		if err != nil {
 			return err
 		}
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			tasks:       tasks,
-		},
-	)
+	return b.printTasks(taskPrintOpts{tasks: tasks})
+}
+
+type taskRerunFailedFlags struct {
+	before string
+	after  string
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedCmd() *cobra.Command {
+	cmd := b.opts.newCmd("retry-failed", b.taskRetryFailedF, true)
+	cmd.Short = "Retry failed runs"
+	cmd.Aliases = []string{"rtf"}
+
+	b.org.register(b.opts.viper, cmd, false)
+	b.globalFlags.registerFlags(b.opts.viper, cmd)
+	registerPrintOptions(b.opts.viper, cmd, &b.taskPrintFlags.hideHeaders, &b.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&b.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.before, "before", "", "runs before this time")
+	cmd.Flags().StringVar(&b.taskRerunFailedFlags.after, "after", "", "runs after this time")
+
+	return cmd
+}
+
+func (b *cmdTaskBuilder) taskRetryFailedF(*cobra.Command, []string) error {
+	if err := b.org.validOrgFlags(&flags); err != nil {
+		return err
+	}
+	tskSvc, _, err := b.svcFn()
+	if err != nil {
+		return err
+	}
+
+	taskIDPresent := b.taskID == ""
+
+	var failedRuns []*influxdb.Run
+	if !taskIDPresent {
+		failedRuns, err = b.getRunsOrg()
+	} else {
+		failedRuns, err = b.getFailedRunsTaskID()
+	}
+	if err != nil {
+		return err
+	}
+
+	for _, run := range failedRuns {
+		newRun, err := tskSvc.RetryRun(context.Background(), run.TaskID, run.ID)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Retry for task %s's run %s queued as run %s.\n", run.TaskID, run.ID, newRun.ID)
+	}
+	return nil
+
+}
+
+// helper function to find all failed runs based on taskID
+func (b *cmdTaskBuilder) getFailedRunsTaskID() ([]*influxdb.Run, error) {
+	// use RunFilter to search for failed runs then return collection of failed tasks
+	tskSvc, _, err := b.svcFn()
+	if err != nil {
+		return nil, err
+	}
+	runFilter := influxdb.RunFilter{}
+	id, err := influxdb.IDFromString(b.taskID)
+	if err != nil {
+		return nil, err
+	}
+	runFilter.Task = *id
+	runFilter.BeforeTime = b.taskRerunFailedFlags.before
+	runFilter.AfterTime = b.taskRerunFailedFlags.after
+	allRuns, _, err := tskSvc.FindRuns(context.Background(), runFilter)
+	if err != nil {
+		return nil, err
+	}
+	var allFailedRuns []*influxdb.Run
+	for _, run := range allRuns {
+		if run.Status == "failed" {
+			allFailedRuns = append(allFailedRuns, run)
+		}
+	}
+	return allFailedRuns, nil
+
+}
+
+// helper function to find all failed runs associated with an organization
+func (b *cmdTaskBuilder) getRunsOrg() ([]*influxdb.Run, error) {
+	// use TaskFilter to get all Tasks in org then search for failed runs in each task.
+	// returns collection of failed tasks
```

Comment:
Same as above re these comments and function name. Definitely update this function name to mention `Failed` runs like the other one at least.